src/components/Api.js

```diff
@@ -0,0 +1,96 @@
+//Autherization code: 196bd4a5-0b68-4944-a5da-dae91687bd99
+export default class Api {
+  constructor(options) {
+    // constructor body
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+  }
+
+  //user api
+
+  updateUserInfo(name, about) {
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        name: name,
+        about: about,
+      }),
+    }).then(this._checkResponse);
+  }
+
+  updateProfileAvatar(avatarUrl) {
+    return fetch(`${this._baseUrl}/users/me/avatar`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        avatar: avatarUrl,
+      }),
+    }).then(this._checkResponse);
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    })
+      .then(this._checkResponse)
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+
+  //card api
+
+  getCardData() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    })
+      .then(this._checkResponse)
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+
+  addNewCard(newCard) {
+    return fetch(`${this._baseUrl}/cards`, {
+      method: "POST",
+      headers: this._headers,
+      body: JSON.stringify(newCard),
+    })
+      .then(this._checkResponse)
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+  deleteCardData(cardId) {
+    return fetch(`${this._baseUrl}/cards/${cardId}`, {
+      method: "DELETE",
+      headers: this._headers,
+    })
+      .then((res) => res.json())
```

Comment:
`      .then((res) => res.json())
`

You should use `.then(this._checkResponse);` in every request because you need to check the respose if it's `ok`